### ISSUE TITLE
Refactor

### ISSUE THREAD

**Issue by mxdxgx** [echolevel/wulfcode#3]
Extract inner classes from wulfcode.java and encapsulate data. 
@echolevel that cool with you ?

**Comment by mxdxgx** [comment]
I'm trying it out on linux, btw. ;)

**Comment by echolevel** [comment]
Great, keen to hear if it works! Sure, any improvements are welcome. I'm experimenting with a Node/Web Audio/Web MIDI version (documenting my progress [here](http://wulfcode.tumblr.com/)), but I aim to keep improving the Java version too.

**Comment by 2bard** [comment]
Would be great to get some unit testing in with the refactoring effort. I've not properly analysed the codebase yet but a few tests wouldn't do any harm.

**Comment by mxdxgx** [comment]
Agreed,I'll add this as a subtask or something (I'm used to jira...lol) :P
